HealthVault/Classes/HTTP/MHVNetworkReachability.m

```diff
@@ -58,76 +38,76 @@ static void HostReachabilityStatusChanged(SCNetworkReachabilityRef target, SCNet
     }
 }
 
-@interface MHVHostReachability (MHVPrivate)
+@interface MHVHostReachability ()
 
--(BOOL) enableNotifications:(BOOL) enable;
--(BOOL) enableCallback:(BOOL) enable;
+@property (nonatomic, assign) SCNetworkReachabilityRef hostRef;
+@property (nonatomic, assign) SCNetworkReachabilityFlags status;
 
 @end
 
 @implementation MHVHostReachability
 
-@synthesize hostName = m_hostName;
-@synthesize status = m_status;
--(BOOL)isReachable
+- (BOOL)hasNetworkConnection
+{
+    return [self isReachable];
+}
+
+- (BOOL)isReachable
 {
-    return ((m_status & kSCNetworkFlagsReachable) != 0 &&
-            (m_status & kSCNetworkFlagsConnectionRequired) == 0
-            );
+    return ((self.status & kSCNetworkFlagsReachable) != 0 &&
+            (self.status & kSCNetworkFlagsConnectionRequired) == 0);
 }
-@synthesize isMonitoring = m_isMonitoring;
 
--(id)initWithUrl:(NSURL *)url
+- (instancetype)initWithUrl:(NSURL *)url
 {
     return [self initWithHostName:url.host];
 }
 
--(id)initWithHostName:(NSString *)hostName
+- (id)initWithHostName:(NSString *)hostName
 {
     MHVCHECK_STRING(hostName);
     
     self = [super init];
-    MHVCHECK_SELF;
-    
-    m_hostName = hostName;
-    
-    const char* szHostName = [hostName cStringUsingEncoding:NSUTF8StringEncoding]; // buffer is owned by NSString
-    MHVCHECK_NOTNULL(szHostName);
-    
-    m_hostRef = SCNetworkReachabilityCreateWithName(NULL, szHostName);
-    MHVCHECK_NOTNULL(m_hostRef);
-    
-    m_status = kSCNetworkFlagsReachable; // Assume the best
-    m_isMonitoring = FALSE;
-    
+    if (self)
+    {
+        _hostName = hostName;
+        
+        const char *szHostName = [hostName cStringUsingEncoding:NSUTF8StringEncoding]; // buffer is owned by NSString
+        
+        _hostRef = SCNetworkReachabilityCreateWithName(NULL, szHostName);
+        _status = kSCNetworkFlagsReachable; // Assume the best
+        _isMonitoring = FALSE;
+        
+        [self startMonitoring];
+    }
     return self;
 }
 
--(void)dealloc
+- (void)dealloc
 {
     [self stopMonitoring];
     
-    if (m_hostRef)
+    if (self.hostRef)
     {
-        CFRelease(m_hostRef);
+        CFRelease(self.hostRef);
     }
-    
 }
 
--(BOOL)refreshStatus
+- (BOOL)refreshStatus
 {
     SCNetworkReachabilityFlags status = 0;
-    if (!SCNetworkReachabilityGetFlags(m_hostRef, &status))
+    
+    if (!SCNetworkReachabilityGetFlags(self.hostRef, &status))
     {
         return FALSE;
     }
     
     return TRUE;
 }
 
--(BOOL)startMonitoring
+- (BOOL)startMonitoring
 {
-    if (m_isMonitoring)
+    if (self.isMonitoring)
```

Comment:
This isn't thread safe. We should fix it if we plan on using this class. The setting and checking of the isMonitoring BOOL needs to be synchronized.